### PR TITLE
Subscription Cleanup Job moved to KEB

### LINKED ISSUE #253: Consider moving k8s jobs to a single component
**Description**

Currently there are the following jobs in the Control Plane:
- Subscription clean up
- Sub account clean up 
- Environments clean up

Subscription clean up job source code is in a separate component whereas two others are part of Kyma Environment Broker source code. Consider moving all the jobs to a single component. 

**Reasons**

In order to make it easier to search in the code base we should store all the jobs in one place. We could use domain oriented approach and put all jobs in KEB.

**Possible solutions**
- move all jobs to KEB <- preferred as it makes dependency management easier
- make separate component for these jobs

### PR DESCRIPTION
**Pull request by akgalwas**
**Description**

Changes proposed in this pull request:

- Moved Subscription Clean Job from separate component to KEB 
- Renamed image to `kyma-environment-subscription-cleanup-job` for consistency with other KEB related jobs
- Fixed regression in Subscription Clean up job (secret binding: failed to list shoots: field label not supported: spec.secretBindingName)

fixes #253

### CODE REVIEW

**Review by koala7659 — COMMENTED** [review]
Please check my comments to Dockerfile and makefile

File: components/kyma-environment-broker/Dockerfile.scj [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+COPY vendor vendor
+
+RUN CGO_ENABLED=0 go build -o /bin/subscriptioncleanup ./cmd/subscriptioncleanup/main.go
+
+# Get latest CA certs
+FROM alpine:latest as certs
+RUN apk --update add ca-certificates
+
+# Final image
+FROM alpine:3.15.0
```
Comment (koala7659) [inline components/kyma-environment-broker/Dockerfile.scj:19]: Please consider using scratch as base image

File: components/kyma-environment-broker/Dockerfile.scj [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+LABEL source = git@github.com:kyma-project/control-plane.git
+
+COPY --from=certs /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
+COPY --from=build /bin/subscriptioncleanup /bin/subscriptioncleanup
+
```
Comment (koala7659) [inline components/kyma-environment-broker/Dockerfile.scj:24]: Please consider using WORKDIR /app convention as it in the previous SCJ Dockerfile

**Review by koala7659 — COMMENTED** [review]
Please see more comments to Dockerfile

File: scripts/generic_make_go.mk [hunk 3]
```diff
@@ -19,6 +19,13 @@ else
 SUBACCOUNT_CLEANUP_IMG_NAME := $(APP_SUBACCOUNT_CLEANUP_NAME)
 endif
 
+# Configuration for Kyma Environment Broker subscription cleanup job image
+ifneq ($(strip $(DOCKER_PUSH_REPOSITORY)),)
+SUBSCRIPTION_CLEANUP_IMG_NAME := $(DOCKER_PUSH_REPOSITORY)$(DOCKER_PUSH_DIRECTORY)/$(APP_SUBSCRIPTION_CLEANUP_NAME)
+else
+CLEANUP_IMG_NAME := $(APP_CLEANUP_NAME)
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply scripts/generic_make_go.mk]: Done.

File: components/kyma-environment-broker/Dockerfile.scj (hunk 1 above)
Comment (akgalwas) [reply components/kyma-environment-broker/Dockerfile.scj]: Doen.

File: components/kyma-environment-broker/Dockerfile.scj [hunk 4]
```diff
@@ -0,0 +1,25 @@
+FROM eu.gcr.io/kyma-project/external/golang:1.17.6-alpine3.15 as build
+
+WORKDIR /go/src/github.com/kyma-project/control-plane/components/kyma-environment-broker
+
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply components/kyma-environment-broker/Dockerfile.scj:4]: Done.

File: components/kyma-environment-broker/Dockerfile.scj (hunk 2 above)
Comment (akgalwas) [reply components/kyma-environment-broker/Dockerfile.scj:23]: Done.

**Comment by koala7659** [comment]
Small comment, you can consider using `v1 ` instead of `v1beta1`. There is a waring `Warning: batch/v1beta1 CronJob is deprecated in v1.21+, unavailable in v1.25+; use batch/v1 CronJob`

**Review by koala7659 — DISMISSED** [review]
LGTM

**Comment by piotrmiskiewicz** [comment]
We are removing Gardener dependency from KEB, see https://github.com/kyma-project/control-plane/pull/1521
Let's wait for it.

**Review by piotrmiskiewicz — APPROVED** [review]

**Review by koala7659 — APPROVED** [review]
lgtm

**Pull request merged**